packages/contracts/src/2.0.0/protocol/Exchange/MixinSignatureValidator.sol

```diff
@@ -257,4 +262,78 @@ contract MixinSignatureValidator is
         // signature was invalid.)
         revert("SIGNATURE_UNSUPPORTED");
     }
+
+    /// @dev Verifies signature using logic defined by Wallet contract.
+    /// @param hash Any 32 byte hash.
+    /// @param walletAddress Address that should have signed the given hash
+    ///                      and defines its own signature verification method.
+    /// @param signature Proof that the hash has been signed by signer.
+    /// @return True if signature is valid for given wallet..
+    function isValidWalletSignature(
+        bytes32 hash,
+        address walletAddress,
+        bytes signature
+    )
+        internal
+        view
+        returns (bool isValid)
+    {
+        bytes memory calldata = abi.encodeWithSelector(
```

Comment:
I think this will bumb the memory pointer unnecessarily, leading to extra gas cost. (not an issue)